elastic/kibana
# Fixes issue with removing spaces from related objects in share to space action

closes #168657

## Summary

Updates the `share to spaces` action to refrain from removing spaces from related objects (objects referenced by the target object).

I have also updated the description of issue #130562, which essentially will replace much of the client-side implementation of this action, to explicitly include this behavior.

### Manual Testing
- Create 2 spaces: A and B
- Add a sample data set (e.g. flight) to space A
- In Discover, create a saved query called "s1" (add a filter that uses the sample data logs data view, and use the filter menu button)
- Go to `Stack Management->Saved` Objects and share the "s1" query to space B
- Verify that the related data view is also shared to space B.
- Un-share the "s1" query from space B
- Verify that the related data view is still shared to space B.

### Automated Tests
- x-pack/plugins/spaces/public/share_saved_objects_to_space/components/share_to_space_flyout_internal.test.tsx
